### PR TITLE
convert spec parameters to snake_case before sending them to ansible.

### PR DESCRIPTION
**Pull request by fabianvf**

### CODE REVIEW

File: pkg/runner/case_conversion.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
+// Based on https://github.com/iancoleman/strcase
+package runner
```
Comment (fabianvf) [inline pkg/runner/case_conversion.go]: Should this move to a util package?

Comment (shawn-hurley) [reply pkg/runner/case_conversion.go]: What about a conversion package or a `paramconv` package(like `strconv` from standard library)?

Comment (fabianvf) [reply pkg/runner/case_conversion.go]: done

**Review by shawn-hurley — APPROVED** [review]
A think a follow up with tests will be good

**Comment by lkhomenk** [comment]
@fabianvf Hi. could you please paste a link to the backgroud issue this PR fixed? at the moment we are facing opposite problems. spec of our CR contains valid k8s affinity which we wanted to reuse in deployments/pod/etc via the our ansible-based operator. But this mandatory MapToSnake is a showstopper for us

**Comment by fabianvf** [comment]
You can still access the unmodified spec of the object in the `_{group}_{kind}` (lower case with dots changed to underscores) variable, so if you have an `apps.example.com/v1alpha1 Sample` object the unmodified spec would be in `_apps_example_com_sample.spec`

**Pull request merged**